test/yaml.cpp

```diff
@@ -38,4 +38,16 @@ TEST_CASE("yaml node conversions", "[yaml_node]") {
 	REQUIRE(decoded_string == string);
 }
 
+TEST_CASE("merge yaml nodes", "[yaml_node]") {
```

Comment:
Would be good to also have a test case for the recursive functionality, so updating a subnode of a subnode. Could be a separate test case, or could be added here.